mdloader_common.h

```diff
@@ -183,6 +184,78 @@ int write_data(int addr, int writesize, int data);
 void list_devices(char *first);
 void strupper(char *str);
 void strlower(char *str);
+void read_user_row(void);
+
+// Smart EEPROM specific
+#define NVMCTRL 0x41004000
+#define NVMCTRL_CTRLA (NVMCTRL)
+#define NVMCTRL_CTRLB (NVMCTRL + 4)
+#define NVMCTRL_ADDR (NVMCTRL + 0x14)
+
+#define NVMCTRL_CTRLA_WMODE_MAN 0x0
+#define NVMCTRL_CTRLB_CMDEX_KEY 0xA5
+#define NVMCTRL_CTRLB_CMD_WQW 0x4
+#define NVMCTRL_CTRLB_CMD_PBC 0x15
+#define NVMCTRL_CTRLB_CMD_EP 0x0
+
+#define NVMCTRL_USER 0x00804000
+
+
+
+typedef union {
+    struct {
+        uint32_t SBLK          : 4; /* bit: 35:32 - Number of NVM Blocks composing a SmartEEPROM sector   */
+        uint32_t PSZ           : 3; /* bit: 38:36 - SmartEEPROM Page Size                                 */
+        uint32_t RAM_ECCDIS    : 1; /* bit:    39 - RAM ECC Disable                                       */
+        uint32_t               : 8; /* bit: 47:40 - Factory settings - do not change                      */
+        uint32_t WDT_ENABLE    : 1; /* bit:    48 - WDT Enable at power-on                                */
+        uint32_t WDT_ALWAYS_ON : 1; /* bit:    49 - WDT Always-On at power-on                             */
+        uint32_t WDT_PERIOD    : 4; /* bit: 53:50 - WDT Period at power-on                                */
+        uint32_t WDT_WINDOW    : 4; /* bit: 57:54 - WDT Window mode time-out at power - on                */
+        uint32_t WDT_EWOFFSET  : 4; /* bit: 61:58 - WDT Early Warning Interrupt Time Offset at power - on */
+        uint32_t WDT_WEN       : 1; /* bit:    62 - WDT Window Mode Enable at power - on                  */
+        uint32_t               : 1; /* bit:    63 - Factory settings - do not change                      */
+    } bit;
+    uint32_t reg;
+} NVMCTRL_USER_ROW_MAPPING1_Type;
```

Comment:
I'm curious how it would be written as, to support big endian, as I haven't done this before.
Still, I think this is beyond the scope of this PR, as you said.